src/python/pants/backend/python/subsystems/python_setup.py

```diff
@@ -23,7 +23,7 @@ class PythonSetup(Subsystem):
   def register_options(cls, register):
     super().register_options(register)
     register('--interpreter-constraints', advanced=True, fingerprint=True, type=list,
-             default=['CPython>=2.7,<3', 'CPython>=3.6,<4'],
+             default=['CPython>=3.6,<4'],
```

Comment:
This is a breaking change for any shop with only python2 available. For them, without ever setting `--interpreter-constraints` things just worked since all their machines had python2.7 only. On upgrade, they break, albeit with a workaround. I think this violates our deprecation policy. Although I'm happy to see this switch I don't think we can really do it without either acknowledging the deviation from the policy, perhaps with a special release notes / release email announcement / slcak #releases announcement. Or else a standard deprecation, adding code that checks if the default constraints are being used, and if so, warns that they'll be switch to this new value.

Reply:
We probably should have made this change sooner with a deprecation cycle. Now that we're only one month away from Python 2 being EOL (I know Pants will still support Py2), I advocate special release notes + release email.

The other breaking change I've been working on is upgrading default Pytest to the 5.x series, which does not support Python 2 (https://github.com/pantsbuild/pants/pull/8648). It would be helpful to make both changes at once and explain both changes in the release notes + release email.

@stuhood @benjyw what do you think?

Reply:
I'm not aware of an existential need to move fast and break things here. Is there something other than the standard desire to have less complicated code that deprecation cycles always push against?

Reply:
> Is there something other than the standard desire to have less complicated code that deprecation cycles always push against motivating this?

Eh there are workarounds, you're right. I wasn't planning on this change until finding in #8646 that some tests failed because it's using Py2 for the subprocess - there are other ways to fix that problem.

I'll update this to use deprecations. Thank you for saying something!